anon/elgam.go

```diff
@@ -0,0 +1,82 @@
+package anon
+
+import (
+	"bytes"
+	"errors"
+	"crypto/cipher"
+	"github.com/dedis/crypto/abstract"
+)
+
+type basicSig struct {
+	C0 abstract.Secret
+	S  abstract.Secret
+}
+
+func signH1preElGam(suite abstract.Suite, message []byte) []byte {
+	H := suite.Hash()
+	H.Write(message)			
+	return H.Sum(nil)
+}
+
+func signH1ElGam(suite abstract.Suite, H1pb []byte, PG abstract.Point) abstract.Secret {
+	H1 := suite.Hash()
+	H1.Write(H1pb)
+	H1.Write(PG.Encode())
+
+	b := H1.Sum(nil)
+	s := suite.Stream(b[:suite.KeyLen()])
+	return suite.Secret().Pick(s)
+}
+
+
+func SignElGam(suite abstract.Suite, random cipher.Stream, message []byte,
```

Comment:
Please add documentation-comments, at least brief ones, before each function (or at least public functions).  

In particular, since this is an implementation of a well-known crypto primitive, include in the comment a pointer (URL) to whatever formulation you used here or found to be most clear.